puppetboard/app.py

```diff
@@ -758,36 +759,42 @@ def query(env):
         select field in the environment block
     :type env: :obj:`string`
     """
-    if app.config['ENABLE_QUERY']:
-        envs = environments()
-        check_env(env, envs)
-
-        form = QueryForm(meta={
-            'csrf_secret': app.config['SECRET_KEY'],
-            'csrf_context': session})
-        if form.validate_on_submit():
-            if form.endpoints.data == 'pql':
-                query = form.query.data
-            elif form.query.data[0] == '[':
-                query = form.query.data
-            else:
-                query = '[{0}]'.format(form.query.data)
-            result = get_or_abort(
-                puppetdb._query,
-                form.endpoints.data,
-                query=query)
-            return render_template('query.html',
-                                   form=form,
-                                   result=result,
-                                   envs=envs,
-                                   current_env=env)
+    if not app.config['ENABLE_QUERY']:
+        log.warn('Access to query interface disabled by administrator.')
+        abort(403)
+
+    envs = environments()
+    check_env(env, envs)
+
+    form = QueryForm(meta={
+        'csrf_secret': app.config['SECRET_KEY'],
+        'csrf_context': session})
+    if form.validate_on_submit():
+        if form.endpoints.data not in ENABLED_QUERY_ENDPOINTS:
+            log.warn('Access to query endpoint %s disabled by administrator.',
+                     form.endpoints.data)
+            abort(403)
+
+        if form.endpoints.data == 'pql':
+            query = form.query.data
+        elif form.query.data[0] == '[':
```

Comment:
Should this perhaps be `form.query.data.strip[0]` instead to correctly handle a query with a leading newline/space? I imagine that could happen pretty easily with a textarea field and copy-paste.

Reply:
@runejuhl thanks for the review. This was pre-existing code (see line 771 on current code), but I'm happy pass it through `strip()` if there is an agreement.

Reply:
> /me toggles the unified diff setting

Sorry about that, you're right of course.

Reply:
I think this can stay as it is. But another PR that makes the existing code more beautiful is always highly appreciated.